ddtrace/propagation/http.py

```diff
@@ -182,10 +181,17 @@ def _extract(headers):
             # Do not fail if the tags are malformed
             try:
                 # We get a Dict[str, str], but need it to be Dict[Union[str, bytes], str] (e.g. _MetaDictType)
-                meta = cast(Dict[Union[str, bytes], str], decode_tagset_string(tags_value))
+                meta = cast(
+                    Dict[Union[str, bytes], str],
+                    {k: v for (k, v) in decode_tagset_string(tags_value) if k in _DatadogMultiHeader._PROPAGATED_TAGS},
```

Comment:
we can take them all, we don't need to filter what we extract.

if it is already in the http headers, there shouldn't be much issue with adding to the span meta.